### ISSUE TITLE
using systemd.exec ProtectHome bugs nexcloud

### ISSUE THREAD

**Issue by Thaodan** [nextcloud/server#15186]
### Steps to reproduce
1. use nextcloud with a php instance running inside a systemd service
 running with ProtectHome=yes
2. use 'cache_path' => '/var/lib/nextcloud/cache',
'datadirectory' => '/var/lib/nextcloud/data',
3. complete the initial setup

### Expected behaviour
Nextcloud should run.

### Actual behaviour
It doesn't because it complains files inside itself are symlinks

### Server configuration

**Operating system**:
Arch Linux
**Web server:**
Nginx + UWSGI
**Database:**
PLSQL
**PHP version:**
7.3.4
**Nextcloud version:** (see Nextcloud admin page)
15.07
**Updated from an older Nextcloud/ownCloud or fresh install:**
fresh
**Where did you install Nextcloud from:**
Distro Package
**Signing status:**

**Logs**
Entries allong the lines of this:
```
alfheim uwsgi[25387]: {"reqId":"pauoaWhaAZVfxvCcEwiH","level":3,"time":"2019-04-16T23:08:48+00:00","remoteAddr":"84.175.160.117","user":"admin","app":"core","method":"POST","url":"\/index.php","message":"Following symlinks is not allowed ('\/var\/lib\/nextcloud\/data\/admin\/.' -> '\/var\/lib\/private\/nextcloud\/data\/admin\/' not inside '\/var\/lib\/nextcloud\/data\/admin\/')","userAgent":"Mozilla\/5.0 (X11; Linux x86_64; rv:66.0) Gecko\/20100101 Firefox\/66.0","version":"15.0.7.0"}
Following symlinks is not allowed ('/var/lib/nextcloud/data/_6164/cache' -> '/var/lib/private/nextcloud/data/_6164/cache/' not inside '/var/lib/nextcloud/data/_6164/')

```

The issue is that systemd hides nextclouds dir  in /var/lib/private/nextcloud gives out /var/lib/nextclous but nextcloud thinks that this is a symlink.
See here for more:
https://www.freedesktop.org/software/systemd/man/systemd.exec.html#ProtectHome=

**Comment by skjnldsv** [comment]
This is a webserver configuration issue.
Following symlink or not is not up to nextcloud for this

**Comment by Thaodan** [comment]
You didn't read the issue nextcloud shouldn't try to do any tricks on the folder to break out of the folder.

**Comment by skjnldsv** [comment]
Hey! So I read your issue but my explanation isn't good, let me try again.
We don't support following symlinks if the webserver is not allowing it, we use realpath to determine the exact file location but we cannot work out of the webserver confinements.

https://github.com/owncloud/core/pull/24899
